R/human_infection.R

```diff
@@ -290,29 +294,52 @@ calculate_treated <- function(
       replace = TRUE
     )
   ])
-
-  successful <- bernoulli_multi_p(parameters$drug_efficacy[drugs])
-  treated_index <- bitset_at(seek_treatment, successful)
-
-  # Update those who have been treated
-  if (treated_index$size() > 0) {
-    variables$state$queue_update('Tr', treated_index)
+  
+  if(parameters$antimalarial_resistance) {
+    resistance_parameters <- get_antimalarial_resistance_parameters(
+      parameters = parameters,
+      drugs = drugs, 
+      timestep = timestep
+    )
+    unsuccessful_treatment_probability <- resistance_parameters$artemisinin_resistance_proportion * resistance_parameters$early_treatment_failure_probability
+    susceptible_to_treatment_index <- bernoulli_multi_p(p = 1 - unsuccessful_treatment_probability)
+    susceptible_to_treatment <- bitset_at(seek_treatment, susceptible_to_treatment_index)
+    drugs <- drugs[susceptible_to_treatment_index]
+    n_early_treatment_failure <- n_treat - susceptible_to_treatment$size()
+    renderer$render('n_early_treatment_failure', n_early_treatment_failure, timestep)
+  } else {
+    susceptible_to_treatment <- seek_treatment
+    n_early_treatment_failure <- n_treat - susceptible_to_treatment$size()
+    renderer$render('n_early_treatment_failure', n_early_treatment_failure, timestep)
+  }
```

Comment:
```suggestion
  } else {
    susceptible_to_treatment <- seek_treatment
    n_early_treatment_failure <- n_treat - susceptible_to_treatment$size()
  }
  renderer$render('n_early_treatment_failure', n_early_treatment_failure, timestep)
```

Can you safely pull this render outside the ifelse here to avoid a duplicated line?

Reply:
Yes - have pulled the lines calculating and rendering n_early_treatment_failure outside of the ifelse statement.